weblate/formats/xwiki.py

```diff
@@ -0,0 +1,195 @@
+"""Specific file formats for XWiki"""
+
+import re
+from copy import deepcopy
+from xml.etree import ElementTree
+from xml.sax.saxutils import escape, unescape
+from django.utils.functional import cached_property
+from translate.storage.properties import xwikifile
+from translate.misc import quote
+from weblate.formats.ttkit import PropertiesFormat, PropertiesUnit
+
+XML_HEADER = """<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+"""
+
+
+class XWikiUnit(PropertiesUnit):
+    """
+        Inspired from PropertiesUnit, allow to override the methods to use the right
+        XWikiDialect methods for decoding properties.
+    """
+
+    @cached_property
+    def source(self):
+        # Need to decode property encoded string
+        return quote.xwiki_properties_decode(super().source)
+
+    @cached_property
+    def target(self):
+        """Return target string from a Translate Toolkit unit."""
+        if self.unit is None:
+            return ""
+        # Need to decode property encoded string
+        # This is basically stolen from
+        # translate.storage.properties.propunit.gettarget
+        # which for some reason does not return translation
+        value = quote.xwiki_properties_decode(self.unit.value)
+        value = re.sub("\\\\ ", " ", value)
+        return value
+
+
+class XWikiPropertiesFormat(PropertiesFormat):
+    """
+    Represents an XWiki Java Properties translation file.
+    This format specification is detailed in
+    https://dev.xwiki.org/xwiki/bin/view/Community/XWiki%20Translations%20Formats/#HXWikiJavaProperties
+    """
+
+    unit_class = XWikiUnit
+    name = 'XWiki Java Properties'
+    format_id = 'xwiki-java-properties'
+    loader = xwikifile
```

Comment:
```suggestion
    loader = ("properties", "xwikifile")
```

This way the import will be done when loading the format and failed import will just disable the file format support. This way the support can be added before translate-toolkit with the changes is released.